vocode/streaming/agent/state_agent.py

```diff
@@ -490,7 +527,20 @@ async def compose_action(self, state):
                 params = parse_llm_dict(response)
         self.logger.info(f"Parsed params: {params}")
 
-        action = self.action_factory.create_action(action_config)
+        try:
+            action = self.action_factory.create_action(action_config)
+        except Exception as e:
+            self.json_transcript.entries.append(
+                StateAgentTranscriptActionError(
+                    action_name=action_name or "action has no name",
```

Comment:
i'm not quite sure, but it's possible at type level and should therefore be handled

one possible cause: the person who created the AI forgot to fill in the action_name field. This will cause all kinds of errors, tbf, but it could be helpful to have one that explicitly calls out the missing data